tests/suites/test_suite_ssl.function

```diff
@@ -1755,3 +1981,76 @@ void ssl_session_serialize_version_check( int corrupt_major,
 
 }
 /* END_CASE */
+
+/* BEGIN_DEPENDENCIES
+ * depends_on:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_PEM_PARSE_C:MBEDTLS_RSA_C:MBEDTLS_ECDSA_C
+ * END_DEPENDENCIES
+ */
+
+/* BEGIN_CASE */
+void test_handshake_state()
+{
+    enum { BUFFSIZE = 1024 };
+    int ret = 0;
+    mbedtls_endpoint ep;
+    mbedtls_endpoint_certificate cert;
+    mbedtls_mock_socket socket; /* dummy socket */
+
+    /* Test basic server initializing */
+
+    TEST_ASSERT( ( ret = mbedtls_endpoint_init( &ep, MBEDTLS_SSL_IS_SERVER ) )
+                 == 0 );
+    TEST_ASSERT( ( ret = mbedtls_endpoint_certificate_init( &ep, &cert ) )
+                 == 0 );
+    TEST_ASSERT( mbedtls_mock_socket_connect( &(ep.socket), &socket, BUFFSIZE )
+                 == 0 );
+
+    /* Test diffrent state of handshake for server */
+
+    TEST_ASSERT( ep.ssl.state == MBEDTLS_SSL_HELLO_REQUEST );
+    TEST_ASSERT( mbedtls_ssl_handshake_step( &(ep.ssl) ) == 0 );
+    TEST_ASSERT( ep.ssl.state == MBEDTLS_SSL_CLIENT_HELLO );
+
+    TEST_ASSERT( mbedtls_endpoint_reinit( &ep, MBEDTLS_SSL_IS_SERVER ) == 0 );
+    TEST_ASSERT( mbedtls_mock_socket_connect( &(ep.socket), &socket, BUFFSIZE ) == 0 );
+    ep.ssl.state = MBEDTLS_SSL_SERVER_NEW_SESSION_TICKET;
```

Comment:
Do we want to assign the state directly or do we want to get there through successive calls to `mbedtls_ssl_handshake_step()`? Does it matter which way we take to get to this state?

Reply:
It depends what do we want to test. I had two thoughts how to test current state of the handshake in this task.

Almost each states of the handshake sequence depends on the previous one and from data received from the second endpoint. If I would like to test some state from the middle of the handshake sequence I would have to do successive calls  mbedtls_ssl_handshake_step for client and server. Then I would have specific state witch could be tested if it is initialized correctly. I was going to do this test in the next task.  

In this tesk I was going to check if mbedtls_ssl_handshake_step called with an incorrectly set state would return an error